test/utils/Deployers.sol

```diff
@@ -83,8 +82,13 @@ contract Deployers is Test {
         }
     }
 
-    function deployFreshManager() internal virtual {
-        manager = new PoolManager(address(this));
+    function deployFreshManager() internal virtual returns (IPoolManager manager_) {
+        bytes memory bytecode =
+            abi.encodePacked(vm.getCode("out/PoolManager.sol/PoolManager.default.json"), abi.encode(address(this)));
```

Comment:
why cant you do it as it was before?

Reply:
I agree that it's somewhat of a problem. But we can probably also migrate periphery to use briefcase instead of v4 directly?
If we deploy the pool manager using the `new` keyword instead of assembly, then the tests run against a version of the pool that is compiled without IR, so the bytecode of the pool manager will be different.

Reply:
Why does `new` not use ir @gretzke ? Didn't know that

Reply:
because of the compilation restriction we set up (`{ paths = "**/test/**", via_ir = false }`). It means that all test files are compiled without IR. If a contract is imported to be deployed using the `new` keyword, it means that this contract is also compiled without IR for the scope of this test and thus tests run against a version compiled without IR. Which is fine in itself but if we want to run tests against the bytecode that will be deployed later on chain we need to avoid importing the contract directly and instead load the bytecode from the file that was compiled with IR.